logic/LogicManager.java

```diff
@@ -130,34 +130,45 @@ private Rule fromStructure(RuleStructure ruleStructure) {
     }
 
     private void validateCyclesThroughNegations(ConceptManager conceptMgr, LogicManager logicMgr) {
+        // Problem: When a derivation of an inferred concept contains its own negation.
+        // Tracking this at runtime is impractical - statically detect & disallow.
+        // Detect: A cycle which passes through atleast one negation.
+        //      Equivalently - a cycle starting from a negated concept in a rule.
```

Comment:
if you want to leave a comment describing this, the habit is to use a 
```
/**
*/
``` 
style comment before the function. This is called a Java-doc comment!

however, since this is just a bug fix of an incorrect implementation, we don't need to leave a comment at all. The namings and variables should be sufficient!
If you want to make it clearer, let's think about how we can further modularise the code to help explain the intent... maybe the inner loop should be its own method called `mayCycle(Rule rule)` or something like that

Reply:
Works.
I guess this PR will show up in the blame if anyone needs an explanation.